### ISSUE TITLE
`nu --lsp` crash with "Char index out of bounds" error

### ISSUE THREAD

**Issue by fredericgiquel** [nushell/nushell#11522]
### Describe the bug

The language server crashes quite often in my use (with `Emacs/lsp-mode` client).

The problem does not seem related to client side and I found an example that can be reliably reproduced.

### How to reproduce

1. Configure your editor to communicate with `nu --lsp`
2. Open the following file (UTF-8 encoding):
```nu
#!/usr/bin/env nu

# Important to reproduce the crash
# use a non ascii char somewhere in comments: è

[a b c d] | filter {
```
3. Check that the lsp server has crashed

Optional:
4. Close the brace at the end of file
5. Restart lsp server (it should not crash)
6. Add an opening brace/parenthesis/quote at the end of file
7. The lsp server should crash again

### Expected behavior

No crash.

### Configuration

| key                | value                                                   |
| ------------------ | ------------------------------------------------------- |
| version            | 0.89.0                                                  |
| branch             |                                                         |
| commit_hash        | 2c1560e281d65a50d90b30c42006755b364b4928                |
| build_os           | linux-x86_64                                            |
| build_target       | x86_64-unknown-linux-gnu                                |
| rust_version       | rustc 1.73.0 (cc66ad468 2023-10-03)                     |
| rust_channel       | 1.73.0-x86_64-unknown-linux-gnu                         |
| cargo_version      | cargo 1.73.0 (9c4383fb5 2023-08-26)                     |
| build_time         | 2024-01-09 20:27:07 +00:00                              |
| build_rust_channel | release                                                 |
| allocator          | mimalloc                                                |
| features           | default, sqlite, static-link-openssl, trash, which, zip |
| installed_plugins  |                                                         |

### Additional context

Backtrace obtained with "RUST_BACKTRACE=full"

```thread 'main' panicked at /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ropey-1.6.1/src/rope.rs:690:41:
called `Result::unwrap()` on an `Err` value: Char index out of bounds: char index 125, Rope/RopeSlice char length 124
stack backtrace:
   0:     0x5593e352f811 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h678bbcf9da6d7d75
   1:     0x5593e297e9ac - core::fmt::write::hb8eaf5a8e45a738e
   2:     0x5593e352a05e - std::io::Write::write_fmt::h9663fe36b2ee08f9
   3:     0x5593e352f5f4 - std::sys_common::backtrace::print::h1360e9450e4f922a
   4:     0x5593e3536853 - std::panicking::default_hook::{{closure}}::h2609fa95cd5ab1f4
   5:     0x5593e353659f - std::panicking::default_hook::h6d75f5747cab6e8d
   6:     0x5593e3536eba - std::panicking::rust_panic_with_hook::h57e78470c47c84de
   7:     0x5593e3536c64 - std::panicking::begin_panic_handler::{{closure}}::h3dfd2453cf356ecb
   8:     0x5593e3535446 - std::sys_common::backtrace::__rust_end_short_backtrace::hdb177d43678e4d7e
   9:     0x5593e35369e2 - rust_begin_unwind
  10:     0x5593e28e2cd3 - core::panicking::panic_fmt::hd1e971d8d7c78e0e
  11:     0x5593e28e30ba - core::result::unwrap_failed::hccb456d39e9c31fc
  12:     0x5593e2ef2eae - ropey::rope::Rope::char_to_line::ha7b636edd27a4ca5
  13:     0x5593e2ef9bbb - nu_lsp::LanguageServer::span_to_range::he6d533fdb48d76bb
  14:     0x5593e2ef71ce - nu_lsp::LanguageServer::serve_requests::h95e4df82785d6619
  15:     0x5593e2a602eb - nu::main::h8d06ff4ef665a1a0
  16:     0x5593e2a6d189 - std::sys_common::backtrace::__rust_begin_short_backtrace::h9ca86ae1cb9ea720
  17:     0x5593e2a6d9d7 - std::rt::lang_start::{{closure}}::h21cd026fb7192bee
  18:     0x5593e352025b - std::rt::lang_start_internal::hc0b4e50f058f62ce
  19:     0x5593e2a62055 - main
  20:     0x7f78f31df6ca - __libc_start_call_main
                               at ./csu/../sysdeps/nptl/libc_start_call_main.h:58:16
  21:     0x7f78f31df785 - __libc_start_main_impl
                               at ./csu/../csu/libc-start.c:360:3
  22:     0x5593e29170ee - _start
  23:                0x0 - <unknown>
```

**Comment by fdncred** [comment]
@schrieveslaach any ideas about this?

**Comment by fredericgiquel** [comment]
Additional info I forgot:

If I remove the `è` char on line 4, the lsp server no longer crashes.

**Comment by schrieveslaach** [comment]
got it fixed, see #11526
